### PR TITLE
test: remove test depend and use vusted

### PR DESCRIPTION
**Pull request by glepnir**
improve test . before lspconfig test depend on neovim test library so it download neovim and build it .now more simple more easy and more fast.

### CODE REVIEW

File: test/lspconfig_spec.lua [hunk 1]
```diff
@@ -1,362 +1,283 @@
...
-          ]])
-
-          ok(exec_lua [[
-            local lspconfig = require("lspconfig")
-            return lspconfig.util.path.is_absolute("../foo/bar") == nil
-          ]])
+          local lspconfig = require 'lspconfig'
+          eq(true, lspconfig.util.path.is_absolute 'foo/bar' == nil)
+
+          eq(true, lspconfig.util.path.is_absolute '../foo/bar' == nil)
```
Comment (dundargoc) [inline test/lspconfig_spec.lua:121]:
```suggestion
          assert.is_nil(lspconfig.util.path.is_absolute '../foo/bar')
```

File: test/lspconfig_spec.lua [hunk 2]
```diff
@@ -1,362 +1,283 @@
...
-            vim.api.nvim_command("cd ../test/test_dir/")
-            local file = vim.fn.getcwd().."/not_exists.txt"
-
-            return lspconfig.util.path.exists(file) == false
-          ]])
+          local lspconfig = require 'lspconfig'
+          -- change the working directory to test directory
+          vim.api.nvim_command 'cd ./test/test_dir/'
+          local file = vim.fn.getcwd() .. '/not_exists.txt'
+          eq(true, lspconfig.util.path.exists(file) == false)
```
Comment (dundargoc) [inline test/lspconfig_spec.lua:54]:
```suggestion
          assert.is_false(lspconfig.util.path.exists(file))
```

**Review by dundargoc — APPROVED** [review]
The CI changes look good.

Took a quick glance at the lua code, I think busted have specific assertions for basically every scenario (`is_nil`, `is_true`, `is_false` etc). Unsure if they're that much readable tbh, but mentioning it in just in case.

Looks good

**Comment by glepnir** [comment]
thanks for review :)

**Pull request merged**